src/components/CardEditor/CardEditForm/AttributesFormItem/AttributesFormItem.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { Add16, Edit16, Delete16 } from '@carbon/icons-react';
+
+import { ComposedModal, Button, List, TextInput } from '../../../../index';
+
+const defaultProps = {
+  value: [],
+  i18n: {},
+};
+
+const propTypes = {
+  /* value */
+  value: PropTypes.arrayOf(PropTypes.shape({})),
+  /* callback when image input value changes (File object) */
+  onChange: PropTypes.func.isRequired,
+  i18n: PropTypes.shape({}),
+};
+
+const AttributesFormItem = ({ value, onChange }) => {
+  const [showEditor, setShowEditor] = useState(false);
+  const [editData, setEditData] = useState({});
+  const [editDataSourceId, setEditDataSourceId] = useState({});
+
+  return (
+    <>
+      {showEditor ? (
+        <ComposedModal
+          header={{
+            label: 'Attribute',
+            title: 'Edit attribute',
+          }}
+          onSubmit={() => {
+            onChange(value.filter(i => i.dataSourceId !== editDataSourceId).concat([editData]));
+            setShowEditor(false);
+            setEditData(null);
+          }}
+          onClose={() => {
+            setShowEditor(false);
+            setEditData(null);
+          }}
+        >
+          <div style={{ paddingBottom: '1rem' }}>
```

Comment:
Can this be swapped to classes vs inline?

Reply:
I've removed these "extra" form items from the PR